src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -825,25 +834,95 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
                     {
                         continue;
                     }
+
                     var better = BetterOperator(candidates[i].Signature, candidates[j].Signature, left, right, ref useSiteDiagnostics);
-                    if (better == BetterResult.Left)
+                    if (better == BetterResult.Left && candidates[j].Kind != OperatorAnalysisResultKind.Worse)
```

Comment:
Given this comment https://github.com/dotnet/roslyn/pull/15839#discussion_r109506143, I would recommend to restore this code (the slow pass) to its exact original state. The risk of messing it up is not worth a possible benefit of saving a couple of CPU cycles.